src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.cs

```diff
@@ -0,0 +1,483 @@
+                project.Solution,
+                nameof(IRemoteExtensionMethodImportCompletionService.GetUnimportedExtensionMethodsAsync),
+                new object[] { document.Id, position, SymbolKey.CreateString(receiverTypeSymbol), namespaceInScope.ToArray(), forceIndexCreation },
+                cancellationToken).ConfigureAwait(false);
+
+            return (serializableItems.ToImmutableArray(), counter);
+        }
+
+        public static async Task<(ImmutableArray<SerializableImportCompletionItem>, StatisticCounter)> GetUnimportedExtensionMethodsInCurrentProcessAsync(
+            Document document,
+            int position,
+            ITypeSymbol receiverTypeSymbol,
+            ISet<string> namespaceInScope,
+            bool forceIndexCreation,
+            CancellationToken cancellationToken)
+        {
+            var counter = new StatisticCounter();
+            var ticks = Environment.TickCount;
+
+            var project = document.Project;
+            var assembly = (await project.GetCompilationAsync(cancellationToken).ConfigureAwait(false))!;
+
+            // Get the metadata name of all the base types and interfaces this type derived from.
+            using var _ = PooledHashSet<string>.GetInstance(out var allTypeNamesBuilder);
+            allTypeNamesBuilder.Add(receiverTypeSymbol.MetadataName);
+            allTypeNamesBuilder.AddRange(receiverTypeSymbol.GetBaseTypes().Select(t => t.MetadataName));
+            allTypeNamesBuilder.AddRange(receiverTypeSymbol.GetAllInterfacesIncludingThis().Select(t => t.MetadataName));
+
+            // interface doesn't inherit from object, but is implicitly convertable to object type.
+            if (receiverTypeSymbol.IsInterfaceType())
+            {
+                allTypeNamesBuilder.Add(nameof(Object));
+            }
+
+            var allTypeNames = allTypeNamesBuilder.ToImmutableArray();
+            var matchedMethods = await GetPossibleExtensionMethodMatchesAsync(
+                project, allTypeNames, forceIndexCreation, isPrecalculation: false, cancellationToken).ConfigureAwait(false);
+
+            counter.GetFilterTicks = Environment.TickCount - ticks;
+            counter.NoFilter = matchedMethods == null;
+
+            // Don't show unimported extension methods if the index isn't ready.
+            if (matchedMethods == null)
+            {
+                // We use a very simple approach to build the cache in the background:
+                // queue a new task only if the previous task is completed, regardless of what
+                // that task is.
+                lock (s_gate)
+                {
+                    if (s_indexingTask.IsCompleted)
+                    {
+                        s_indexingTask = Task.Run(() => GetPossibleExtensionMethodMatchesAsync(project, allTypeNames, forceIndexCreation: false, isPrecalculation: true, CancellationToken.None));
```

Comment:
why are we passing `forceIndexCreation: false` here?

Reply:
fixed